### PR TITLE
[RS-1446] - WAF UI needs permission to delete applicationlayers resources

### PR DESCRIPTION
**Pull request by mikestephen**
## Description

There's a bug in the WAF UI currently where the user needs to be able to delete applicationlayers resources if they disable the feature, but the tigera-network-admin user does not have permission to do so, so the call fails.
This change adds that permission to tigera-network-admin.

This will need to be cherry-picked to the correct branch afterwards to be included in calient 3.18.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
